src/React/Basic/Hooks/Aff.purs

```diff
@@ -37,11 +38,33 @@ useAff ::
   deps ->
   Aff a ->
   Hook (UseAff deps a) (Maybe a)
-useAff deps aff =
+useAff = useAff' (const Nothing)
+
+--| A variant of `useAff` where the asynchronous effect's result is preserved up
+--| until the next run of the asynchronous effect _completes_.
+--|
+--| Contrast this with `useAff`, where the asynchronous effect's result is
+--| preserved only up until the next run of the asynchronous effect _starts_.
+useSteppingAff ::
+  forall deps a.
+  Eq deps =>
+  deps ->
+  Aff a ->
+  Hook (UseAff deps a) (Maybe a)
+useSteppingAff = useAff' identity
+
+useAff' ::
+  forall deps a.
+  Eq deps =>
+  (Maybe (Either Error a) -> Maybe (Either Error a)) ->
+  deps ->
+  Aff a ->
+  Hook (UseAff deps a) (Maybe a)
+useAff' initUpdater deps aff =
```

Comment:
Do you think it would make sense to export `useAff'` as well? Contrived example, but I could imagine maybe wanting something other than `identity` or `const Nothing` if wanting to track loading state along with the `Aff`'s result (to preserve and render the previous result but disable a button while some request is in flight, e.g.)
```haskell
useReloadingAff
  :: forall deps a
   . Eq deps
  => deps
  -> Aff a
  -> Hook
       (UseAff deps { isLoading :: Boolean, result :: a })
       (Maybe { isLoading :: Boolean, result :: a })
useReloadingAff deps aff =
  React.useAff' initUpdater deps aff'
  where
  initUpdater = (map <<< map) (_ { isLoading = true })
  aff' = map { isLoading: false, result: _ } aff
```

Reply:
I'd want to have a pretty convincing use case. It's easy enough to copy/paste for current edge cases, so I worry it's not worth exposing that complexity.

Reply:
But happy to defer to others since I'm not an active user right now.

Same goes for merging/publishing this.

Reply:
> It's easy enough to copy/paste for current edge cases, so I worry it's not worth exposing that complexity.

Keeping the external API simple makes sense to me 👍 In cases where I've wanted anything other than the `Maybe a` state that `useAff` returns I'd just use a separate `useState` and use `useAff` for scheduling. So for the snippet I had above I think I'd probably just do
```haskell
useReloadingAff deps aff = React.do
  state /\ setState <- useState Nothing
  useAff deps do
    liftEffect $ setState (map (_ { isLoading = true }))
    result <- aff
    liftEffect $ setState \_ -> Just { isLoading: false, result }
  pure state
```
which is not much more complicated than the version using `useAff'`.

I'm not really an active user either aside from side projects. Lumi's use case for `useStepperAff` seems like enough motivation for merging this as-is, but I'll leave this open for feedback for a couple more days before merging.